src/transformers/models/textnet/convert_textnet_original_to_pytorch.py

```diff
@@ -0,0 +1,202 @@
+        stage_config = config_dict[f"stage{stage_ix}"]
+
+        merged_dict = {}
+
+        # Iterate through the list of dictionaries
+        for layer in stage_config:
+            for key, value in layer.items():
+                if key != "name":
+                    # Check if the key is already in the merged_dict
+                    if key in merged_dict:
+                        merged_dict[key].append(value)
+                    else:
+                        # If the key is not in merged_dict, create a new list with the value
+                        merged_dict[key] = [value]
+        backbone_config[f"stage{stage_ix}"] = merged_dict
+
+    neck_in_channels = []
+    neck_out_channels = []
+    neck_kernel_size = []
+    neck_stride = []
+    neck_dilation = []
+    neck_groups = []
+
+    for i in range(1, 5):
+        layer_key = f"reduce_layer{i}"
+        layer_dict = config_dict["neck"].get(layer_key)
+
+        if layer_dict:
+            # Append values to the corresponding lists
+            neck_in_channels.append(layer_dict["in_channels"])
+            neck_out_channels.append(layer_dict["out_channels"])
+            neck_kernel_size.append(layer_dict["kernel_size"])
+            neck_stride.append(layer_dict["stride"])
+            neck_dilation.append(layer_dict["dilation"])
+            neck_groups.append(layer_dict["groups"])
+
+    textnet_config = TextNetConfig(
+        stem_kernel_size=config_dict["first_conv"]["kernel_size"],
+        stem_stride=config_dict["first_conv"]["stride"],
+        stem_num_channels=config_dict["first_conv"]["in_channels"],
+        stem_out_channels=config_dict["first_conv"]["out_channels"],
+        stem_act_func=config_dict["first_conv"]["act_func"],
+        conv_layer_kernel_sizes=[
+            backbone_config["stage1"]["kernel_size"],
+            backbone_config["stage2"]["kernel_size"],
+            backbone_config["stage3"]["kernel_size"],
+            backbone_config["stage4"]["kernel_size"],
+        ],
+        conv_layer_strides=[
+            backbone_config["stage1"]["stride"],
+            backbone_config["stage2"]["stride"],
+            backbone_config["stage3"]["stride"],
+            backbone_config["stage4"]["stride"],
+        ],
+        hidden_sizes=[
+            config_dict["first_conv"]["out_channels"],
+            backbone_config["stage1"]["out_channels"][-1],
+            backbone_config["stage2"]["out_channels"][-1],
+            backbone_config["stage3"]["out_channels"][-1],
+            backbone_config["stage4"]["out_channels"][-1],
+        ],
+        out_features=["stage1", "stage2", "stage3", "stage4"],
+        out_indices=[1, 2, 3, 4],
+        image_size=size,
+    )
+
+    return textnet_config
+
+
+def convert_textnet_checkpoint(checkpoint_url, checkpoint_config_url, pytorch_dump_folder_path):
+    response = requests.get(checkpoint_config_url)
+    content = response.text
+    namespace = {}
+
+    exec(content, namespace)
```

Comment:
Agree, But this config url is a python file, ([example](https://raw.githubusercontent.com/czczup/FAST/main/config/fast/ic17mlt/fast_base_ic17mlt_640.py)) . Is there any other safer way to do this ?

Reply:
@amyeroberts  Gentle remainder .

Reply:
I have made this change.

Reply:
It doesn't seem to have been pushed?

Reply:
What I have done is, push the config file (python file) into repo and download it using hf_hub_download. My understanding is when file is downloaded from hub, it will be safer to exec. Let me know if my understanding is wrong and this has to fixed in some other way.

Reply:
@amyeroberts  I have removed the dynamic execution, I have preprocessed all the python config files into a json files. Hope this approach is okay.

Reply:
Also the original_pixel_values has been prepared and stored in hub, avoid performing preprocessing in conversion script.